anesthetic/samples.py

```diff
@@ -1171,8 +1171,41 @@ def logL_P(self, nsamples=None, beta=None):
 
     logL_P.__doc__ += _logZ_function_shape
 
+    def contour(self, logL=None):
+        """Convert contour from (index or None) to a float loglikelihood.
+
+        Convention is that live points are inclusive of the contour.
+
+        Helper function for:
+            - NestedSamples.live_points,
+            - NestedSamples.dead_points,
+            - NestedSamples.truncate.
+
+        Parameters
+        ----------
+        logL : float or int, optional
+            Loglikelihood or iteration number
+            If not provided, return the contour containing the last set of
+            live points.
+
+        Returns
+        -------
+        logL : float
+            Loglikelihood of contour
+        """
+        if logL is None:
+            logL = self.loc[self.logL > self.logL_birth.max()].logL.iloc[0]
+        elif isinstance(logL, float):
+            pass
+        else:
+            try:
+                logL = float(self.logL[logL])
+            except KeyError:
+                pass
+        return logL
+
```

Comment:
This is not covered by the tests -- I agree that your additional float check is safer than what was there before. Can we think of anything which the try except covers (given that it is now never hit by the tests?) if not, remove the extra code, and just move to straight return statements from the if elif else block.